apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -758,17 +742,13 @@ func (api *ProvisionerAPI) Constraints(ctx context.Context, args params.Entities
 	}
 	for i, entity := range args.Entities {
 		tag, err := names.ParseMachineTag(entity.Tag)
-		if err != nil {
+		if err != nil || !canAccess(tag) {
 			result.Results[i].Error = apiservererrors.ServerError(apiservererrors.ErrPerm)
 			continue
 		}
-		machine, err := api.getMachine(canAccess, tag)
+		cons, err := api.machineService.GetMachineConstraints(ctx, coremachine.Name(tag.Id()))
 		if err == nil {
-			var cons constraints.Value
-			cons, err = machine.Constraints()
-			if err == nil {
-				result.Results[i].Constraints = cons
-			}
+			result.Results[i].Constraints = cons
 		}
 		result.Results[i].Error = apiservererrors.ServerError(err)
```

Comment:
Invert these please (i.e. `if err != nil` should be the check)

Reply:
Done.